src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -202,6 +202,8 @@ private void SetGlobalErrorIfTrue(bool arg)
                     new LocalizableResourceString(messageResourceName, CodeAnalysisResources.ResourceManager, typeof(CodeAnalysisResources)));
             }
 
+            AddCircularStructDiagnostics(compilation.GlobalNamespace);
```

Comment:
>AddCircularStructDiagnostics [](http://example.com/codeflow?start=12&length=28)

Is this the only call site? Consider making this a local function then.

Reply:
Fixed